hazelcast/include/hazelcast/client/proxy/SerializingProxy.h

```diff
@@ -373,7 +373,7 @@ class HAZELCAST_API SerializingProxy
     }
 
     template<typename T>
-    typename std::enable_if<std::is_trivial<T>::value, boost::future<T>>::
```

Comment:
I don't think the changes in this file have any effect because nobody tried to use this `decode` function with a boost::uuid. No objection to the change though.